client/hosting/server-settings/main.tsx

```diff
@@ -290,7 +290,8 @@ const ServerSettings = ( { fetchUpdatedData }: ServerSettingsProps ) => {
 	 * Otherwise, we show the activation notice, which may be empty.
 	 */
 	const shouldShowUpgradeBanner =
-		! hasAtomicFeature || ( ! hasTransfer && ! hasSftpFeature && ! isWpcomStagingSite );
+		! isLoadingSftpData &&
```

Comment:
I have noticed that with the proposed change we are losing the upgrade banner completely for eCommerce trial sites:

![Markup on 2024-11-07 at 15:18:27](https://github.com/user-attachments/assets/fff96050-c5c6-49c9-b131-821c47c7a8ab)

To address this, we could consider checking for `isECommerceTrial` here as well - or - adjust the logic that calculates `isLoadingSftpData`.

My apologies for not noticing this in the first review comment.

Reply:
Yeah the logic here is pretty tangled 🧶, I see why the eCommerce trial sites aren't showing the banner now, because it doesn't have the SSH/SFTP features, thanks for catching this 👍

Reply:
Thanks for the fix, Candy! All is working well now. The change is looking good and the banner is displaying on eCommerce trial sites again. 🙂